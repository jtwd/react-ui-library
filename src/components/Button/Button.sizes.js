import spacers from '../_theme/spacers'
import fontSizes from '../_theme/fontSizes'

const sizes = {
  default: `
    margin: 0 ${spacers.xxs} ${spacers.md};
    padding: 0.66rem ${spacers.md};
    font-size: ${fontSizes.md};
  `,
  lg: `
    margin: 0 ${spacers.xs} ${spacers.md};
    padding: ${spacers.lg} ${spacers.xl};
    font-size: ${fontSizes.lg};
  `,
  sm: `
    margin: 0 ${spacers.xxs} ${spacers.sm};
    padding: .33rem ${spacers.sm};
    font-size: ${fontSizes.sm};
  `,
  iconOnly: `
    padding: 0.38rem ${spacers.sm}; 
  `,
  iconOnlySm: `
    padding: ${spacers.xxs} ${spacers.xxs};
  `,
  iconOnlyLg: `
    padding: .85rem ${spacers.md};
  `
}

export default sizes
