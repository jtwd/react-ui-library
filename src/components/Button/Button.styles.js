import styled, { css } from 'styled-components'
import { switchProp, ifProp } from 'styled-tools'
import { Link } from 'react-router-dom'

import { themeProps, mapProp } from '../_theme/utils'
import trimChildren from '../_theme/mixins/trimChildren'
import boxShadowAnimation from '../_theme/mixins/boxShadowAnimation'
import theme from './Button.theme'

const t = themeProps(theme)

export const StyledButton = styled.button`
  border-radius: ${t.borderRadius};
  font-family: ${t.fontFamily};
  font-weight: ${t.fontWeight};
  line-height: ${t.lineHeight};
  ${props => t.uiStyle[props.uiStyle].boxShadow && boxShadowAnimation('all')}
  
  font-size: ${switchProp('uiSize', mapProp('fontSize', t.uiSize))};
  margin: ${switchProp('uiSize', mapProp('margin', t.uiSize))};
  padding: ${ifProp({iconOnly: true},
              switchProp('uiSize', mapProp('iconOnlyPadding', t.uiSize)),
              switchProp('uiSize', mapProp('padding', t.uiSize))
          )};

  background-color: ${switchProp('uiStyle', mapProp('backgroundColor', t.uiStyle))};
  color: ${switchProp('uiStyle', mapProp('color', t.uiStyle))};
  border: ${switchProp('uiStyle', mapProp('border', t.uiStyle))};

  &:focus,
  &:hover {
    background-color: ${switchProp('uiStyle', mapProp('hoverBackgroundColor', t.uiStyle))};
    color: ${switchProp('uiStyle', mapProp('hoverColor', t.uiStyle))};
    border: ${switchProp('uiStyle', mapProp('hoverBorder', t.uiStyle))};
    ${props => props.uiStyle !== 'link' && css`box-shadow: ${t.hoverBoxShadow} ${t.boxShadowColor};`}
    text-decoration: none;
  }

  cursor: pointer;
  display: inline-block;
  min-height: 1rem;
  min-width: 2rem;
  vertical-align: middle;
  text-decoration: none;

  ${trimChildren('hor')};
`

export const IconWrap = styled.span`
  left: ${ifProp({ iconOnly: true },
          switchProp('uiSize', mapProp('iconOnlyLeft', t.uiSize)),
          switchProp('uiSize', mapProp('iconLeft', t.uiSize))
        )};
  top: ${ifProp({ iconOnly: true },
          switchProp('uiSize', mapProp('iconOnlyTop', t.uiSize)),
          switchProp('uiSize', mapProp('iconTop', t.uiSize))
        )};
  margin: ${ifProp({ iconOnly: true },
            '0',
            switchProp('uiSize', mapProp('iconMargin', t.uiSize))
          )};
          
  display: inline-block;
  position: relative;
  vertical-align: middle;
`

export const LinkButton = styled(Link)`
  display: inline-block;
`

export const AButton = styled.a``
