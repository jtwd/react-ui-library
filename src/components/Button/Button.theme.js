import { getTheme } from '../_theme/utils'
import * as c from '../_theme/palette'

const { fontSizes, spacing, radius, lineHeights, transitions, fonts } = getTheme()

export default {
  default: {
    borderRadius: radius.rounded,
    trainstion: transitions.hover,
    fontFamily: fonts.primary.name,
    fontWeight: fonts.primary.bold,
    lineHeight: lineHeights.tight,
    uiSize: {
      default: {
        fontSize: fontSizes.md,
        margin: `0 ${spacing.xxs} ${spacing.md}`,
        padding: `${spacing.sm} ${spacing.md}`,
        iconOnlyPadding: `0.45rem ${spacing.md}`,
        iconOnlyLeft: '0',
        iconOnlyTop: '2px',
        iconLeft: '-.33rem',
        iconTop: '-.25rem',
        iconMargin: '0 0 -.66rem'
      },
      small: {
        fontSize: fontSizes.sm,
        margin: `0 ${spacing.xxs} ${spacing.sm}`,
        padding: `${spacing.xs} ${spacing.sm}`,
        iconOnlyPadding: `0.42rem ${spacing.sm}`,
        iconLeft: '-.25rem',
        iconTop: '0.06rem',
        iconMargin: '-.2rem 0'
      },
      large: {
        fontSize: fontSizes.lg,
        margin: `0 ${spacing.xs} ${spacing.md}`,
        padding: `${spacing.md} ${spacing.lg}`,
        iconOnlyPadding: `.72rem ${spacing.lg}`,
        iconLeft: `-${spacing.xs}`,
        iconTop: '0rem',
        iconMargin: `-.48rem 0`
      }
    },
    uiStyle: {
      default: {
        border: '0',
        color: c.light(),
        backgroundColor: c.tertiary(),
        hoverColor: c.light(),
        hoverBackgroundColor: c.tertiary('dark'),
        hoverBorder: '0',
        boxShadow: true
      },
      primary: {
        border: '0',
        color: c.secondary(),
        backgroundColor: c.primary(),
        hoverColor: c.secondary('dark'),
        hoverBackgroundColor: c.primary('dark'),
        hoverBorder: '0',
        boxShadow: true
      },
      secondary: {
        border: '0',
        color: c.primary(),
        backgroundColor: c.secondary(),
        hoverColor: c.primary('dark'),
        hoverBackgroundColor: c.secondary('dark'),
        hoverBorder: '0',
        boxShadow: true
      },
      danger: {
        border: '0',
        color: c.light(),
        backgroundColor: c.alert(),
        hoverColor: c.light(),
        hoverBackgroundColor: c.alert('dark'),
        hoverBorder: '0',
        boxShadow: true
      },
      link: {
        backgroundColor: 'transparent',
        color: c.primary(),
        hoverBackgroundColor: 'transparent',
        hoverColor: c.light(),
        border: '0',
        hoverBorder: '0',
        boxShadow: false
      }
    }
  }
}
