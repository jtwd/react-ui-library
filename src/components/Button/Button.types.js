import { white, grey, darkGrey, blue, darkBlue, yellow, lightYellow, darkYellow, red, darkRed } from '../_theme/colors'

const types = {
  default: `
    color: ${white};
    background: ${grey};
    &:hover, &:focus {
      background: ${darkGrey};
    }
  `,
  primary: `
    background: ${blue};
    color: ${yellow};
    &:hover, &:focus {
      background: ${darkBlue};
      color: ${lightYellow}
    }
  `,
  secondary: `
    background: ${yellow};
    color: ${blue};
    &:hover, &:focus {
      background: ${darkYellow};
      color: ${darkBlue}
    }
  `,
  danger: `
    background: ${red};
    color: ${white};
    &:hover, &:focus {
      background: ${darkRed};
      color: ${white}
    }
  `
}

export default types
