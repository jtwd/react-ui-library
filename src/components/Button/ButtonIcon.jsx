import React from 'react'
import { string, bool } from 'prop-types'
import styled from 'styled-components'
import { ifProp, switchProp } from 'styled-tools'

import Icon from '../Icon'
import { themeProps, mapProp } from '../_theme/utils'
import theme from './Button.theme'

const t = themeProps(theme)

const StyledSpan = styled.span`
  left: ${ifProp({ iconOnly: true }, 
          switchProp('uiSize', mapProp('iconOnlyLeft', t.uiSize)), 
          switchProp('uiSize', mapProp('iconLeft', t.uiSize))
        )};
  top: ${ifProp({ iconOnly: true }, 
          switchProp('uiSize', mapProp('iconOnlyTop', t.uiSize)), 
          switchProp('uiSize', mapProp('iconTop', t.uiSize))
        )};
  margin: ${ifProp({ iconOnly: true }, 
            '0', 
            switchProp('uiSize', mapProp('iconMargin', t.uiSize))
          )};
          
  display: inline-block;
  position: relative;
  vertical-align: middle;
`

const StyledIcon = ({iconOnly, uiStyle, uiSize, icon, ...props}) =>
  <StyledSpan iconOnly={iconOnly} uiSize={uiSize} uiStyle={uiStyle}>
    <Icon icon={icon} {...props} />
  </StyledSpan>

StyledIcon.propTypes = {
  icon: string.isRequired,
  uiStyle: string,
  uiSize: string,
  iconOnly: bool
}

StyledIcon.defaultProps = {
  iconOnly: false,
  uiStyle: 'default',
  uiSize: 'default'
}

const ButtonIcon = ({icon, iconOnly, large, small}) =>
  <StyledIcon icon={icon} iconOnly={iconOnly} large={large} small={small} />

ButtonIcon.propTypes = {
  icon: string.isRequired,
  iconOnly: bool,
  large: bool,
  small: bool
}

ButtonIcon.defaultProps = {
  iconOnly: false,
  large: false,
  small: false
}

export default ButtonIcon
