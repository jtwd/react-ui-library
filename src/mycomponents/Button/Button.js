import styled from 'styled-components'
import { bool } from 'prop-types'
import { radiusRound, transTextHover, primaryNormal, lineHeightTight } from '../../theme/units'
import { trimChildrenHorizontal } from '../../theme/utils/utils.mixins'
import getPropStylesFromTheme from '../../theme/utils/getPropStylesFromTheme'
import types from './Button.types'
import sizes from './Button.sizes'

const tm = {
  borderRadius: radiusRound,
  transition: transTextHover,
  weight: primaryNormal,
  lineHeight: lineHeightTight,
  types,
  sizes
}

const Button = styled.button`
  border-radius: ${tm.borderRadius};
  transition: ${tm.transition};
  font-weight: ${tm.weight};
  line-height: ${tm.lineHeight};
  border: 0;
  cursor: pointer;
  display: inline-block;
  min-height: 1rem;
  min-width: 3rem;
  vertical-align: baseline;

  ${props => !props.large && !props.small && getPropStylesFromTheme(tm, 'sizes', 'default')} ${props =>
      props.small && getPropStylesFromTheme(tm, 'sizes', 'sm')} ${props =>
      props.large && getPropStylesFromTheme(tm, 'sizes', 'lg')} ${props =>
      !props.primary && !props.secondary && getPropStylesFromTheme(tm, 'types', 'default')} ${props =>
      props.primary && getPropStylesFromTheme(tm, 'types', 'primary')} ${props =>
      props.secondary && getPropStylesFromTheme(tm, 'types', 'secondary')} ${trimChildrenHorizontal};
`

Button.propTypes = {
  primary: bool,
  secondary: bool,
  large: bool,
  small: bool
}

export default Button
