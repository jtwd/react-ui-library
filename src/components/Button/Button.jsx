import React from 'react'
import styled from 'styled-components'
import { bool, node, string } from 'prop-types'
import { radiusRound, transTextHover, primaryNormal, lineHeightTight } from '../../theme/units'
import trimChildren from '../../theme/mixins/trimChildren'
import getPropStylesFromTheme from '../../theme/utils/getPropStylesFromTheme'
import getIcon from '../../theme/utils/utils.icons'
import types from './Button.types'
import sizes from './Button.sizes'
import ButtonIcon from './ButtonIcon'

const tm = {
  borderRadius: radiusRound,
  transition: transTextHover,
  weight: primaryNormal,
  lineHeight: lineHeightTight,
  types,
  sizes
}

const StyledButton = styled.button`
  border-radius: ${tm.borderRadius};
  transition: ${tm.transition};
  font-weight: ${tm.weight};
  line-height: ${tm.lineHeight};
  border: 0;
  cursor: pointer;
  display: inline-block;
  min-height: 1rem;
  min-width: 3rem;
  vertical-align: middle;

  ${props => !props.large && !props.small && getPropStylesFromTheme(tm, 'sizes')}
  
  ${props => props.small && getPropStylesFromTheme(tm, 'sizes', 'sm')}
   
  ${props => props.large && getPropStylesFromTheme(tm, 'sizes', 'lg')}
  
  ${props => !props.large && !props.small && props.iconOnly && getPropStylesFromTheme(tm, 'sizes', 'iconOnly')}
  
  ${props => props.small && props.iconOnly && getPropStylesFromTheme(tm, 'sizes', 'iconOnlySm')}
   
  ${props => props.large && props.iconOnly && getPropStylesFromTheme(tm, 'sizes', 'iconOnlyLg')}
   
  ${props => !props.primary && !props.secondary && getPropStylesFromTheme(tm, 'types', 'default')}
   
  ${props => props.primary && getPropStylesFromTheme(tm, 'types', 'primary')}
   
  ${props => props.secondary && getPropStylesFromTheme(tm, 'types', 'secondary')}
    
  ${trimChildren('hor')};
`

function Button({ primary, secondary, large, small, icon, children }) {
  const iconOnly = (children === null)
  const validIcon = (getIcon(icon) !== null)

  if(iconOnly && !validIcon) return null // don't show if there is no valid contents
  return (
    <StyledButton primary={primary} secondary={secondary} large={large} small={small} iconOnly={iconOnly}>
      {icon && <ButtonIcon iconOnly={iconOnly} icon={icon} large={large} small={small} />}
      {children}
    </StyledButton>
  )
}

Button.propTypes = {
  /** Type of button */
  primary: bool,
  /** Type of button */
  secondary: bool,
  /** Size of button */
  large: bool,
  /** Size of button */
  small: bool,
  /** Name of Icon */
  icon: string,
  /** Contents of button */
  children: node
}

Button.defaultProps = {
  primary: false,
  secondary: false,
  large: false,
  small: false,
  icon: '',
  children: null
}

export default Button
