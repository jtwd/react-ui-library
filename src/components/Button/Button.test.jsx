import React from 'react'
import {shallow} from 'enzyme'

import Button from './Button'

describe('Button component', () => {
  it('renders a default button by default', () => {
    const component = shallow(<Button>Default</Button>)
    expect(component).toMatchSnapshot()
  })

  it('renders a primary button when passed prop', () => {
    const component = shallow(<Button primary>Primary</Button>)
    expect(component).toMatchSnapshot()
  })

  it('renders a secondary button when passed prop', () => {
    const component = shallow(<Button secondary>Primary</Button>)
    expect(component).toMatchSnapshot()
  })

  it('renders a danger button when passed prop', () => {
    const component = shallow(<Button danger>Danger</Button>)
    expect(component).toMatchSnapshot()
  })

  it('renders a link button when passed prop', () => {
    const component = shallow(<Button link>Link</Button>)
    expect(component).toMatchSnapshot()
  })

  it('renders a large button when passed prop', () => {
    const component = shallow(<Button large>Primary</Button>)
    expect(component).toMatchSnapshot()
  })

  it('renders a small button when passed prop', () => {
    const component = shallow(<Button small>Primary</Button>)
    expect(component).toMatchSnapshot()
  })

  it('renders a large, primary button when passed prop', () => {
    const component = shallow(<Button primary large>Primary</Button>)
    expect(component).toMatchSnapshot()
  })

  it('renders a small, secondary button when passed prop', () => {
    const component = shallow(<Button small secondary>Primary</Button>)
    expect(component).toMatchSnapshot()
  })

  it('should render an icon if a valid icon is passed', () => {
    const component = shallow(<Button icon="close">Icon Button</Button>)
    expect(component).toMatchSnapshot()
  })

  it('should render a primary button with an icon', () => {
    const component = shallow(<Button primary icon="close">Icon Button</Button>)
    expect(component).toMatchSnapshot()
  })

  it('should render a secondary button with an icon', () => {
    const component = shallow(<Button secondary icon="close">Icon Button</Button>)
    expect(component).toMatchSnapshot()
  })

  it('should render a large button with an icon', () => {
    const component = shallow(<Button large icon="close">Icon Button</Button>)
    expect(component).toMatchSnapshot()
  })

  it('should render a small button with an icon', () => {
    const component = shallow(<Button small icon="close">Icon Button</Button>)
    expect(component).toMatchSnapshot()
  })

  it('should render a correct icon only button if no children are passed with an icon', () => {
    const component = shallow(<Button icon="close" />)
    expect(component).toMatchSnapshot()
  })

  it('should render a correct icon only large button if no children are passed with an icon and large prop', () => {
    const component = shallow(<Button icon="close" large />)
    expect(component).toMatchSnapshot()
  })

  it('should render a correct icon only small button if no children are passed with an icon and small prop', () => {
    const component = shallow(<Button icon="close" small />)
    expect(component).toMatchSnapshot()
  })

  it('should render a correct icon only primary button if no children are passed with an icon and primary prop', () => {
    const component = shallow(<Button icon="close" primary />)
    expect(component).toMatchSnapshot()
  })

  it('should render a correct icon only secondary button if no children are passed with an icon and secondary prop', () => {
    const component = shallow(<Button icon="close" secondary />)
    expect(component).toMatchSnapshot()
  })

  it('should still render a normal button if icon prop is not valid and children is supplied', () => {
    const component = shallow(<Button icon="qwerty">Invalid icon</Button>)
    expect(component).toMatchSnapshot()
  })

  it('should not render if icon prop is not valid and no children are supplied', () => {
    const component = shallow(<Button icon="qwerty" />)
    expect(component).toMatchSnapshot()
  })
})
