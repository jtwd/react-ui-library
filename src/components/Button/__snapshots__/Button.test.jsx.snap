// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders a danger button when passed prop 1`] = `
<styled.button
  danger={true}
  iconOnly={false}
  large={false}
  link={false}
  primary={false}
  secondary={false}
  small={false}
  type="button"
>
  Danger
</styled.button>
`;

exports[`Button component renders a default button by default 1`] = `
<styled.button
  danger={false}
  iconOnly={false}
  large={false}
  link={false}
  primary={false}
  secondary={false}
  small={false}
  type="button"
>
  Default
</styled.button>
`;

exports[`Button component renders a large button when passed prop 1`] = `
<styled.button
  danger={false}
  iconOnly={false}
  large={true}
  link={false}
  primary={false}
  secondary={false}
  small={false}
  type="button"
>
  Primary
</styled.button>
`;

exports[`Button component renders a large, primary button when passed prop 1`] = `
<styled.button
  danger={false}
  iconOnly={false}
  large={true}
  link={false}
  primary={true}
  secondary={false}
  small={false}
  type="button"
>
  Primary
</styled.button>
`;

exports[`Button component renders a link button when passed prop 1`] = `
<styled.button
  danger={false}
  iconOnly={false}
  large={false}
  link={true}
  primary={false}
  secondary={false}
  small={false}
  type="button"
>
  Link
</styled.button>
`;

exports[`Button component renders a primary button when passed prop 1`] = `
<styled.button
  danger={false}
  iconOnly={false}
  large={false}
  link={false}
  primary={true}
  secondary={false}
  small={false}
  type="button"
>
  Primary
</styled.button>
`;

exports[`Button component renders a secondary button when passed prop 1`] = `
<styled.button
  danger={false}
  iconOnly={false}
  large={false}
  link={false}
  primary={false}
  secondary={true}
  small={false}
  type="button"
>
  Primary
</styled.button>
`;

exports[`Button component renders a small button when passed prop 1`] = `
<styled.button
  danger={false}
  iconOnly={false}
  large={false}
  link={false}
  primary={false}
  secondary={false}
  small={true}
  type="button"
>
  Primary
</styled.button>
`;

exports[`Button component renders a small, secondary button when passed prop 1`] = `
<styled.button
  danger={false}
  iconOnly={false}
  large={false}
  link={false}
  primary={false}
  secondary={true}
  small={true}
  type="button"
>
  Primary
</styled.button>
`;

exports[`Button component should not render if icon prop is not valid and no children are supplied 1`] = `null`;

exports[`Button component should render a correct icon only button if no children are passed with an icon 1`] = `
<styled.button
  danger={false}
  iconOnly={true}
  large={false}
  link={false}
  primary={false}
  secondary={false}
  small={false}
  type="button"
>
  <ButtonIcon
    icon="close"
    iconOnly={true}
    large={false}
    small={false}
  />
</styled.button>
`;

exports[`Button component should render a correct icon only large button if no children are passed with an icon and large prop 1`] = `
<styled.button
  danger={false}
  iconOnly={true}
  large={true}
  link={false}
  primary={false}
  secondary={false}
  small={false}
  type="button"
>
  <ButtonIcon
    icon="close"
    iconOnly={true}
    large={true}
    small={false}
  />
</styled.button>
`;

exports[`Button component should render a correct icon only primary button if no children are passed with an icon and primary prop 1`] = `
<styled.button
  danger={false}
  iconOnly={true}
  large={false}
  link={false}
  primary={true}
  secondary={false}
  small={false}
  type="button"
>
  <ButtonIcon
    icon="close"
    iconOnly={true}
    large={false}
    small={false}
  />
</styled.button>
`;

exports[`Button component should render a correct icon only secondary button if no children are passed with an icon and secondary prop 1`] = `
<styled.button
  danger={false}
  iconOnly={true}
  large={false}
  link={false}
  primary={false}
  secondary={true}
  small={false}
  type="button"
>
  <ButtonIcon
    icon="close"
    iconOnly={true}
    large={false}
    small={false}
  />
</styled.button>
`;

exports[`Button component should render a correct icon only small button if no children are passed with an icon and small prop 1`] = `
<styled.button
  danger={false}
  iconOnly={true}
  large={false}
  link={false}
  primary={false}
  secondary={false}
  small={true}
  type="button"
>
  <ButtonIcon
    icon="close"
    iconOnly={true}
    large={false}
    small={true}
  />
</styled.button>
`;

exports[`Button component should render a disabled button 1`] = `
<styled.button
  danger={false}
  disabled={true}
  iconOnly={false}
  large={false}
  link={false}
  primary={false}
  secondary={false}
  small={false}
  type="button"
>
  Disabled
</styled.button>
`;

exports[`Button component should render a large button with an icon 1`] = `
<styled.button
  danger={false}
  iconOnly={false}
  large={true}
  link={false}
  primary={false}
  secondary={false}
  small={false}
  type="button"
>
  <ButtonIcon
    icon="close"
    iconOnly={false}
    large={true}
    small={false}
  />
  Icon Button
</styled.button>
`;

exports[`Button component should render a primary button with an icon 1`] = `
<styled.button
  danger={false}
  iconOnly={false}
  large={false}
  link={false}
  primary={true}
  secondary={false}
  small={false}
  type="button"
>
  <ButtonIcon
    icon="close"
    iconOnly={false}
    large={false}
    small={false}
  />
  Icon Button
</styled.button>
`;

exports[`Button component should render a secondary button with an icon 1`] = `
<styled.button
  danger={false}
  iconOnly={false}
  large={false}
  link={false}
  primary={false}
  secondary={true}
  small={false}
  type="button"
>
  <ButtonIcon
    icon="close"
    iconOnly={false}
    large={false}
    small={false}
  />
  Icon Button
</styled.button>
`;

exports[`Button component should render a small button with an icon 1`] = `
<styled.button
  danger={false}
  iconOnly={false}
  large={false}
  link={false}
  primary={false}
  secondary={false}
  small={true}
  type="button"
>
  <ButtonIcon
    icon="close"
    iconOnly={false}
    large={false}
    small={true}
  />
  Icon Button
</styled.button>
`;

exports[`Button component should render an icon if a valid icon is passed 1`] = `
<styled.button
  danger={false}
  iconOnly={false}
  large={false}
  link={false}
  primary={false}
  secondary={false}
  small={false}
  type="button"
>
  <ButtonIcon
    icon="close"
    iconOnly={false}
    large={false}
    small={false}
  />
  Icon Button
</styled.button>
`;

exports[`Button component should still render a normal button if icon prop is not valid and children is supplied 1`] = `
<styled.button
  danger={false}
  iconOnly={false}
  large={false}
  link={false}
  primary={false}
  secondary={false}
  small={false}
  type="button"
>
  <ButtonIcon
    icon="qwerty"
    iconOnly={false}
    large={false}
    small={false}
  />
  Invalid icon
</styled.button>
`;
